otelib/abstractfilter.py

```diff
@@ -1,51 +1,51 @@
-from abc import ABC, abstractmethod
 import requests
 import json
-from models import *
-from pipe import *
-from apierror import *
+from abc import ABC, abstractmethod
+
+from .apierror import ApiError
+from .pipe import Pipe
+from .settings import Settings
```

Comment:
```suggestion
from otelib.apierror import ApiError
from otelib.pipe import Pipe
from otelib.settings import Settings
```

Reply:
What is the reason for this suggestion? I think the relative imports are safer. If the user for some reason has another package called otelib installed, this change will import from that package instead of the modules in the current directory.

Reply:
Relative imports are by definition unstable. It's like using relative paths. It only works in a very specific situation.

People should install the `otelib` package (if we end up calling it this when publishing). They should not have other packages with this name. This will be a unique dedicated package. The argument you're posing is a user-based setup issue and not one we can take into account when developing Python packages for general distribution.

Reply:
Hmm, I am still not convinced... Both absolute and relative imports will break if you move files around. I see no difference in robustness in that respect. But I agree that my argument that the absolute path may lead to conflicts with other packages is not a good argument, since you are specifying the full path from the root package. However, that has the effect that oteapi must be properly installed on your system before you can test it. Relative imports does not have this requirement.

Reply:
> Both absolute and relative imports will break if you move files around.

Definitely not. Only the target files. For relative imports - if you move the file around that is doing the importing it will fail. For absolute imports this is not true.

> I see no difference in robustness in that respect. But I agree that my argument that the absolute path may lead to conflicts with other packages is not a good argument, since you are specifying the full path from the root package. However, that has the effect that oteapi must be properly installed on your system before you can test it. Relative imports does not have this requirement.

Neither does absolute imports as demonstrated in oteapi-services.
And oteapi must be properly installed either way - but I understand you meant otelib. And yes - I'd always go for a package being installed as the default expectation - otherwise there is no idea to creating a package. Then we might as well have created a series of scripts.